nautobot/extras/models/groups.py

```diff
@@ -313,13 +316,78 @@ def get_queryset(self):
 
     @property
     def members(self):
-        """Return the member objects for this group."""
+        """Return the member objects for this group, never cached."""
         # If there are child groups, return the generated group queryset, otherwise use this group's
         # `filter` directly.
         if self.children.exists():
             return self.get_group_queryset()
         return self.get_queryset()
 
+    @property
+    def members_cached(self):
+        """Return the member objects for this group, cached if available."""
+        cache_key = f"{self.__class__.__name__}.{self.id}.cached_members"
+
+        unpickled_query = None
+        try:
+            cached_query = cache.get(cache_key)
+            if cached_query is not None:
+                unpickled_query = pickle.loads(cached_query)
+        except pickle.UnpicklingError:
+            logger.warning("Failed to unpickle cached members for %s", self)
+        finally:
+            if unpickled_query is None:
+                unpickled_query = self.members.all()
+                cached_query = pickle.dumps(unpickled_query)  # Explicitly pickle the query to evaluate it.
+                cache.set(cache_key, cached_query, get_settings_or_config("DYNAMIC_GROUPS_MEMBER_CACHE_TIMEOUT"))
+
+        return unpickled_query
+
+    def get_members(self, skip_cache=False, force_update_cache=False):
+        """
+        Return the member objects for this group.
+
+        Args:
+            skip_cache (bool): Whether to skip the cache and run the query directly. Optional. Default: False.
+            force_update_cache (bool): Whether to force an update of the cache. Optional. Default: False. Ignored if `skip_cache` is True.
+        """
+
+        if skip_cache:
+            return self.members
+
+        if force_update_cache:
```

Comment:
More concerns about having these as two separate params - it's not obvious without reading the code that `force_update_cache` is completely ignored if `skip_cache==True`, for example, and again I wonder why you'd ever want to call `.get_members(skip_cache=True)` instead of just `.members`?